mm2src/mm2_main/src/rpc.rs

```diff
@@ -269,26 +271,48 @@ async fn rpc_service(req: Request<Body>, ctx_h: u32, client: SocketAddr) -> Resp
     // https://github.com/artemii235/SuperNET/issues/219
     let rpc_cors = match ctx.conf["rpccors"].as_str() {
         Some(s) => try_sf!(HeaderValue::from_str(s)),
-        None => HeaderValue::from_static("http://localhost:3000"),
+        None => {
+            if ctx.is_https() {
+                HeaderValue::from_static("https://localhost:3000")
+            } else {
+                HeaderValue::from_static("http://localhost:3000")
+            }
+        },
     };
 
     // Convert the native Hyper stream into a portable stream of `Bytes`.
     let (req, req_body) = req.into_parts();
+
+    if req.method == Method::OPTIONS {
+        return Response::builder()
+            .status(StatusCode::OK)
+            .header(ACCESS_CONTROL_ALLOW_ORIGIN, rpc_cors)
+            .header("Access-Control-Allow-Methods", "POST, OPTIONS")
+            .header("Access-Control-Allow-Headers", "Content-Type")
+            .header("Access-Control-Max-Age", "3600")
+            .body(Body::empty())
+            .unwrap();
+    }
+
     let req_bytes = try_sf!(hyper::body::to_bytes(req_body).await, ACCESS_CONTROL_ALLOW_ORIGIN => rpc_cors);
     let req_str = String::from_utf8_lossy(req_bytes.as_ref());
-    let is_invalid_input = req_str.chars().any(|c| c == '<' || c == '>' || c == '&');
+    let is_invalid_input = req_str.chars().any(|c| NON_ALLOWED_CHARS.contains(&c));
     if is_invalid_input {
         return Response::builder()
             .status(500)
+            .header(ACCESS_CONTROL_ALLOW_ORIGIN, rpc_cors)
             .header(CONTENT_TYPE, APPLICATION_JSON)
-            .body(Body::from(err_to_rpc_json_string("Invalid input")))
+            .body(Body::from(err_to_rpc_json_string(&format!(
+                "Invalid input: contains one or more of the following non-allowed characters: {:?}",
+                NON_ALLOWED_CHARS
+            ))))
             .unwrap();
     }
     let req_json: Json = try_sf!(json::from_slice(&req_bytes), ACCESS_CONTROL_ALLOW_ORIGIN => rpc_cors);
```

Comment:
we can scope this for early release

```rust
    let req_json: Json = {
        let req_bytes = try_sf!(hyper::body::to_bytes(req_body).await, ACCESS_CONTROL_ALLOW_ORIGIN => rpc_cors);
        let req_str = String::from_utf8_lossy(req_bytes.as_ref());
        let is_invalid_input = req_str.chars().any(|c| NON_ALLOWED_CHARS.contains(&c));
        if is_invalid_input {
            return Response::builder()
                .status(500)
                .header(ACCESS_CONTROL_ALLOW_ORIGIN, rpc_cors)
                .header(CONTENT_TYPE, APPLICATION_JSON)
                .body(Body::from(err_to_rpc_json_string(&format!(
                    "Invalid input: contains one or more of the following non-allowed characters: {:?}",
                    NON_ALLOWED_CHARS
                ))))
                .unwrap();
        };
        try_sf!(json::from_slice(&req_bytes), ACCESS_CONTROL_ALLOW_ORIGIN => rpc_cors)
    };
```

Reply:
Thanks for the suggestion, fixed here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2191/commits/570bd7964836d3a075578690e44c0cfe0fe8da6f